core/src/test/java/org/wso2/carbon/transports/TransportManagerTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ *  Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.wso2.carbon.transports;
+
+import org.testng.Assert;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+import org.wso2.carbon.transports.transporter.CustomCarbonTransport;
+
+public class TransportManagerTest {
+
+    private TransportManager transportManager;
+    private CustomCarbonTransport carbonTransport;
+    private CustomCarbonTransport carbonTransport2;
+
+    public TransportManagerTest() {
+    }
+
+    @BeforeTest
+    public void setup() {
+        transportManager = new TransportManager();
+        carbonTransport = new CustomCarbonTransport("dummyId");
+        carbonTransport2 = new CustomCarbonTransport("dummyId2");
+        transportManager.registerTransport(carbonTransport);
+        transportManager.registerTransport(carbonTransport2);
+    }
+
+    @Test
+    public void testUnsuccessfullStartTransport() {
+        try {
+            transportManager.startTransport("wrongId");
+        } catch (IllegalArgumentException e) {
+            String exceptionMessage = "wrongId not found";
+            Assert.assertEquals(exceptionMessage, e.getMessage());
+        }
+    }
+
+
+    @Test(dependsOnMethods = {"testUnsuccessfullStartTransport"})
+    public void testSuccessfullStartTransport() {
+        try {
+            transportManager.startTransport("dummyId");
+        } catch (IllegalArgumentException e) {
+            Assert.assertFalse(true);
+        }
+        Assert.assertTrue(true);
+    }
+
+    @Test(dependsOnMethods = {"testSuccessfullStartTransport"})
+    public void testUnsuccessfullStopTransport() {
+        try {
+            transportManager.stopTransport("wrongId");
+        } catch (IllegalArgumentException e) {
+            String exceptionMessage = "wrongId not found";
+            Assert.assertEquals(exceptionMessage, e.getMessage());
+        }
+    }
+
+    @Test(dependsOnMethods = {"testUnsuccessfullStopTransport"})
+    public void testSuccessfullStopTransport() {
+        try {
+            transportManager.stopTransport("dummyId");
+        } catch (IllegalArgumentException e) {
+            Assert.assertTrue(false);
```

Comment:
If test throws an exception on success, then yes, we can use expectexception annotation. But if exception is thrown in this test, then the test should fail.